libc/src/__support/FPUtil/FPBits.h

```diff
@@ -127,7 +127,18 @@ template <> struct FPLayout<FPType::IEEE754_Binary128> {
 };
 
 template <> struct FPLayout<FPType::X86_Binary80> {
+// x86_64
+#if __SIZEOF_LONG_DOUBLE__ == 16
   using StorageType = UInt128;
+// i386-linux-gnu
+#elif __SIZEOF_LONG_DOUBLE__ == 12
+  using StorageType = UInt<__SIZEOF_LONG_DOUBLE__ * CHAR_BIT>;
+#else
+# TODO: https://github.com/llvm/llvm-project/issues/115184
+# Android i386 uses `long double == double` i.e. `sizeof(long double) == 8`
```

Comment:
I think you should add extra android i386 condition to https://github.com/llvm/llvm-project/blob/main/libc/src/__support/macros/properties/types.h#L24

Reply:
For android, `#define __LDBL_MANT_DIG__ 53`: https://godbolt.org/z/M61Edn5c4, which falls into the first existing case.